src/lib/multikey_dict.py

```diff
@@ -26,6 +26,17 @@ def __init__(self, num_of_keys: int):
         self.keys_to_multikeys = {}
         self.data = {}
 
+    def __eq__(self, value):
+        if isinstance(value, MultiKeyDict):
```

Comment:
Use a guard clause first.

if not isinstance(value, MultiKeyDict):
  return False
...